pandas/core/ops.py

```diff
@@ -1494,18 +1494,20 @@ def _construct_result(left, result, index, name, dtype=None):
     not be enough; we still need to override the name attribute.
     """
     out = left._constructor(result, index=index, dtype=dtype)
-
-    out.name = name
+    out.__finalize__(left)
+    if name is None:
+        out.name = name
```

Comment:
hmm, this i pretty odd, what exactly is the issue here?

Reply:
I have realized that the `if` condition is not correct here. Assign `name` to `out.name` directly should be ok.

Fixed now.